### ISSUE TITLE
Impove accessibility and make the book downloadable as PDF document

### ISSUE THREAD

**Issue by opeolluwa** [loftwah/linux-for-pirates#9]
1. Integrate [MDbook](https://rust-lang.github.io/mdBook/) to optimize the document navigation and make it downloadable as a PDF document
2. Include a workflow to build the book on updates and fixes

**Comment by loftwah** [comment]
You are fantastic. Thank you for getting involved. I'll sort this side of the project out soon so issues and PRs can be adequately managed with GitHub Actions. 🙇‍♂️🌟

**Comment by loftwah** [comment]
Have you implemented this before? Would you be able to create a branch and show me how it is installed? If not I can look at it over the weekend. I'll read the PDF that you emailed tonight too.

**Comment by opeolluwa** [comment]
I've implemented it before

**Comment by opeolluwa** [comment]
> Have you implemented this before? Would you be able to create a branch and show me how it is installed? If not I can look at it over the weekend. I'll read the PDF that you emailed tonight too.

I will also create a branch to explain how to set up the in a development environment

**Comment by loftwah** [comment]
Thank you.

**Comment by loftwah** [comment]
mdbook is a binary, is there any way it can be implemented into a Git repo without me having to write my own Action? If not I can look for something else.

**Comment by opeolluwa** [comment]
A GitHub action can be implemented to build the pages and push it to `gh-branch`, I'll send a sample

**Comment by loftwah** [comment]
Cool we may as well write one then :) I was hoping for something that already existed.

**Comment by opeolluwa** [comment]
Here is the link to an Mdbook action from one of my projects [https://github.com/opeolluwa/molybdenum/blob/master/.github/workflows/deploy.yml](https://github.com/opeolluwa/molybdenum/blob/master/.github/workflows/deploy.yml)

**Comment by loftwah** [comment]
```bash
Error: -16 22:45:18 [ERROR] (mdbook::utils): Error: Couldn't open SUMMARY.md in "/home/runner/work/linux-for-pirates/linux-for-pirates/src" directory
Error: -16 22:45:18 [ERROR] (mdbook::utils): 	Caused By: No such file or directory (os error 2)
Error: Process completed with exit code 101.
```

I'm sure I'll figure this out soon

**Comment by loftwah** [comment]
I'm not really liking mdBook so far. I've spent about half an hour trying to get this going and it keeps failing.

**Comment by loftwah** [comment]
I might find something else when the time is more appropriate unless they can give us something that works right away. I could have been spending this time actually writing the book. Great idea but I'm not sure mdBook is mature enough for me to adopt it here.

**Comment by loftwah** [comment]
Can use the same issue but for now, mdBook is a no-go.

**Comment by opeolluwa** [comment]
Hmmm 🤔 

> ```shell
> Error: -16 22:45:18 [ERROR] (mdbook::utils): Error: Couldn't open SUMMARY.md in "/home/runner/work/linux-for-pirates/Linux-for-pirates/src" directory
> Error: -16 22:45:18 [ERROR] (mdbook::utils): 	Caused By: No such file or directory (os error 2)
> Error: Process completed with exit code 101.
> ```
> 
> I'm sure I'll figure this out soon

It's probably something with the Setup

**Comment by loftwah** [comment]
I won't get the capacity to look at this again for another two-three weeks. mdBook really need to improve their documentation if people are going to adopt it in their projects.

**Comment by opeolluwa** [comment]
I've been using Mdbook for almost a year and I haven't had any issue with it. That is the same reason I recommended it at first.

**Comment by opeolluwa** [comment]
Please see [this](https://opeolluwa.github.io/molybdenum/introduction.html)  for a project I was using it for.

**Comment by loftwah** [comment]
Your project has a slightly different folder structure and includes `docs/book.toml` for what I assume is configuration. I didn't see a single mention of anything like that in what I read. This isn't on you :) Don't worry.

**Comment by opeolluwa** [comment]
> I won't get the capacity to look at this again for another two-three weeks. mdBook really need to improve their documentation if people are going to adopt it in their projects.

I'm quite engaged too, if you wouldn't mind, let's see in 3 weeks to troubleshoot the error. In the meantime, you can recommend alternatives

**Comment by loftwah** [comment]
I spotted that too, and commented out the `cd docs` line. If you'd like to sure thing and I'll get to it when I can. I was hoping to have this sorted for you too.

**Comment by loftwah** [comment]
If it were me, I'd be reaching for an alternative because once I've lost time on a project, I tend never to revisit it ever again.

**Comment by loftwah** [comment]
[bookdown](https://bookdown.org/yihui/bookdown/github.html) looks alright.

I'm not familiar with R language

**Comment by opeolluwa** [comment]
> [bookdown](https://bookdown.org/yihui/bookdown/github.html) looks alright.

Yeah! Just checked it out

**Comment by opeolluwa** [comment]
@loftwah We sure can switch to it.

**Comment by opeolluwa** [comment]
I'd love to get to the source of the glitch once I'm free. At least for the fun of it. 😄

I'm the meantime, let's just switch to another static site builder

**Comment by loftwah** [comment]
I've reverted back to the default with Jekyll. I managed to get mdBook built by adding sudo to the `apt-get` commands but that isn't something recommended by GitHub Actions and it ended up failing again on the same thing anyway. What does this even give us anyway? What is the difference between something like that and [print-friendly](https://www.printfriendly.com/)? I've always just used Markdown and been done with it.

EDIT: It is going to bug me all day until I fix it now hehe
